src/TransloaditClient.js

```diff
@@ -563,7 +691,49 @@ class TransloaditClient {
     })
 
     if (method === 'post' || method === 'put' || method === 'del') {
-      this._appendForm(req, opts.params, opts.fields)
+      const extraData = Object.assign(
+        {tus_num_expected_upload_files: opts.tus_num_expected_upload_files},
+        opts.fields
+      )
+      this._appendForm(req, opts.params, extraData)
+    }
+  }
+
+  _sendTusRequest (opts, cb, onProgress) {
+    let uploadsDone = 0
+    const streamLabels = Object.keys(this._tus_streams)
+    const tlClient = this
+    onProgress = onProgress || (() => {})
+    for (const label of streamLabels) {
+      const file = this._tus_streams[label]
+      const filename = file.path ? path.basename(file.path) : label
+      const tusUpload = new tus.Upload(file, {
+        endpoint: opts.tus_url,
+        resume: true,
+        metadata: {
+          assembly_url: opts.assembly_ssl_url,
+          fieldname: label,
+          filename
+        },
+        uploadSize: fs.statSync(file.path).size, // todo this will fail for streams without path
+        onError: cb,
+        onProgress(bytesUploaded, bytesTotal) {
+          onProgress({
+            uploadProgress: {bytesUploaded, bytesTotal, file: label}
```

Comment:
I think it would be more helpful to provide progress for the entire assembly and not individual files, similar as Uppy does. What do you think?

Reply:
@Acconut done